lib/manageiq/api/client/acts_as_ar_query.rb

```diff
@@ -0,0 +1,175 @@
+module ManageIQ
+  module API
+    class Client
+      class ActsAsArQuery
+        include Enumerable
+        attr_accessor :klass, :options
+
+        SORT_ASC = "asc".freeze
+        SORT_DESC = "desc".freeze
+
+        def initialize(klass, opts = nil)
+          @klass = klass
+          @options = opts || {}
+        end
+
+        def where(*val)
+          conditional(:where, val)
+        end
+
+        def not(*val)
+          conditional(:not, val)
+        end
+
+        def order(*val)
+          options[:order] ||= {}
+          val.flatten.each do |sval|
+            case sval
+            when String
+              sval.split(",").each do |value|
+                sort_attr, sort_order = value.split(" ")
+                options[:order][sort_attr.to_s] = sort_order || SORT_ASC
+              end
```

Comment:
wow, this is getting advanced

The old code tried to do as little manipulation as possible.
Since it was delegating to AR - best to leave the logic there.

Guess the new use case is that we are implementing the AR interface (via http), so this change is probably necessary.